scu_builders.py

```
@@ -0,0 +1,335 @@
+    lines_per_file = max(lines_per_file, 1)
+
+    start_line = 0
+    file_number = 0
+
+    # These do not vary throughout the loop
+    output_folder = abs_main_folder + "/scu/"
+    output_filename_prefix = "scu_" + out_filename
+
+    # Clear out any existing files (usually we will be overwriting,
+    # but we want to remove any that are pre-existing that will not be
+    # overwritten, so as to not compile anything stale)
+    clear_out_existing_files(output_folder, extension)
+
+    for file_count in range(0, num_output_files):
+        end_line = start_line + lines_per_file
+
+        # special case to cover rounding error in final file
+        if file_count == (num_output_files - 1):
+            end_line = len(found_includes)
+
+        write_output_file(
+            file_count, found_includes, start_line, end_line, output_folder, output_filename_prefix, extension
+        )
+
+        start_line = end_line
+
+    # Write the exceptions each in their own scu gen file,
+    # so they can effectively compile in "old style / normal build".
+    for exception_count in range(len(found_exceptions)):
+        write_exception_output_file(
+            exception_count, found_exceptions[exception_count], output_folder, output_filename_prefix, extension
+        )
+
+
+def generate_scu_files(verbose, is_release_build):
+
+    print("=============================")
+    print("Single Compilation Unit Build")
+    print("=============================")
+    print("Generating SCU build files")
+    global _verbose
+    _verbose = verbose
+    global _is_release_build
+    _is_release_build = is_release_build
+
+    # check we are running from the correct folder
+    curr_folder = os.path.abspath("./")
+    # parent_path = basename(normpath(curr_folder))
+
+    if folder_not_found("core") or folder_not_found("platform") or folder_not_found("scene"):
+        print("ERROR - scu_builders.py must be run from the godot folder.")
```

better to raise an exception in case of error, this will bubble up and crash the script (i.e. return a non-zero process status and print the stacktrace)
Currently this silently fail and scons continue happily which most likely produce compilation crashes later on that are then hard to pinpoint

---

On top of that, you shouldn't need to require a given folder to run from by using `__file__` variable that gives you the path of the current script

---

Just do `raise RuntimeError("scu_builders.py must be run from the godot folder.")` instead of the print